0_intro_to_ml_using_jax.ipynb

```diff
@@ -0,0 +1,3758 @@
+{
```

Comment:
<p>Should we explain why this is? (I'm not sure but I would guess that JAX saw that the output of the impure_print_side_effect function does not change based on the return of the print function)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='92'/>

Reply:
<p>Yes, your reason is right. We do mention in the next cell that changing shape triggers a retrace. Do you think that is explicit enough?  </p>